index.js

```diff
@@ -0,0 +1,29 @@
+// index.js
+protobuf = require("protobufjs");
+
+// Object.freeze(Object.prototype);
+
+let person1 = {firstName:"John", lastName:"Doe", age:50, eyeColor:"blue"};
+let person2 = {firstCoolName:"John", lastName:"Doe", age:50, eyeColor:"blue"};
+
+// Vector 1
+let evilkey = "__proto__.firstName"
+let evilval = "evilvalue"
+protobuf.util.setProperty(person1, evilkey, evilval); 
+
+// Vector 2
+let obj = new protobuf.ReflectionObject("Test")
+obj.setParsedOption({}, evilval, evilkey);
+
+// Vector 3
+let p = `option (foo).__proto__.someprop= "somevalue";` 
+protobuf.parse(p)
+
+// Vector 4
+protobuf.load("/path/to/untrusted.proto", function(err, root) { return });
```

Comment:
[comment]: <> (FrogbotReview Applicable 23 1 23 74 "protobuf.load(\"/path/to/untrusted.proto\", function(err, root) { return })" /index.js)

## ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableHighSeverity.png)<br>    high Applicable dependency CVE Vulnerability 📦🔍
	
Finding: Prototype pollution `Object.freeze` remediation was not detected, The vulnerable function protobufjs.parse is called with external input, The vulnerable function protobufjs.load is called

### 👇 Details

#### Description
	
The scanner checks whether any of the following vulnerable functions are called:

* `util.setProperty` with external input to its 2nd (`path`) or 3rd (`value`) arguments.
* `ReflectionObject.setParsedOption` with external input to its 2nd (`name`) or 3rd (`value`) arguments.
* `parse` with external input to its 1st (`source`) argument.
* `load`	

#### Cve details

The package protobufjs before 6.11.3 are vulnerable to Prototype Pollution which can allow an attacker to add/modify properties of the Object.prototype. This vulnerability can occur in multiple ways: 1. by providing untrusted user input to util.setProperty or to ReflectionObject.setParsedOption functions 2. by parsing/loading .proto files

[[🐸 JFrog Frogbot]](https://github.com/jfrog/frogbot#readme)